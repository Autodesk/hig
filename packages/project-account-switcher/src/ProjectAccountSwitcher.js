import React, { Component } from "react";
import PropTypes from "prop-types";

import ProjectAccountSwitcherPresenter from "./presenters/ProjectAccountSwitcherPresenter";
import ProjectAccountSwitcherBehavior from "./behavior/ProjectAccountSwitcherBehavior";

export default class ProjectAccountSwitcher extends Component {
  static propTypes = {
    /** Heading title for the list of Accounts */
    accountTitle: PropTypes.string,
    /** List of Accounts */
    accounts: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.string,
        image: PropTypes.string,
        label: PropTypes.string
      })
    ),
    /** Currently selected Account */
    activeAccount: PropTypes.shape({
      id: PropTypes.string,
      image: PropTypes.string,
      label: PropTypes.string
    }),
    /** Currently selected Project */
    activeProject: PropTypes.shape({
      id: PropTypes.string,
      image: PropTypes.string,
      label: PropTypes.string
    }),
    /** Called when the selected Account or the selected Project changes, the new selected Account and Project will be passed as parameters: (account, project) */
    onChange: PropTypes.func,
    /** Called when the selected Account changes, the new selected Account will be passed as parameter */
    onAccountChange: PropTypes.func,
    /** Called when the selected Project changes, the new selected Project will be passed as parameter */
    onProjectChange: PropTypes.func,
    /** Called when a user clicks on the element */
    onClick: PropTypes.func,
    /** Called when a user clicks on the target element */
    onTargetClick: PropTypes.func,
    /** List of Projects */
    projects: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.string,
        image: PropTypes.string,
        label: PropTypes.string
      })
    ),
    /** Heading title for the list of Projects */
    projectTitle: PropTypes.string
  };

  render() {
    const {
      accounts,
      accountTitle,
      projects,
      projectTitle,
      onChange,
      onAccountChange,
      onProjectChange,
      onClick,
      onTargetClick
    } = this.props;

    return (
      <ProjectAccountSwitcherBehavior
        accounts={accounts}
        activeAccount={this.props.activeAccount}
        activeProject={this.props.activeProject}
        onChange={onChange}
        onAccountChange={onAccountChange}
        onProjectChange={onProjectChange}
        onClick={onClick}
        onTargetClick={onTargetClick}
        projects={projects}
      >
        {({
          activeAccount,
          activeProject,
          handleAccountClick,
          handleProjectClick,
          handleTargetClick,
          open
        }) => (
          <ProjectAccountSwitcherPresenter
            accounts={accounts}
            accountTitle={accountTitle}
            activeAccount={activeAccount}
            activeProject={activeProject}
            onAccountClick={handleAccountClick}
            onProjectClick={handleProjectClick}
            onTargetClick={handleTargetClick}
            open={open}
            projects={projects}
            projectTitle={projectTitle}
          />
        )}
      </ProjectAccountSwitcherBehavior>
    );
  }
}
