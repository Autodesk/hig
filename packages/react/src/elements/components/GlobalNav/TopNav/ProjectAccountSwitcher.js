/**
 * ProjectAccountSwitcher
 *
 *
 *
*/

import React, { Component } from "react";
import PropTypes from "prop-types";

import ProjectAccountSwitcherAdapter from "../../../../adapters/GlobalNav/TopNav/ProjectAccountSwitcherAdapter"; // eslint-disable-line max-len
import Project from "./Project";
import Account from "./Account";

class ProjectAccountSwitcher extends Component {
  static propTypes = {
    accounts: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.string,
        image: PropTypes.string,
        label: PropTypes.string
      })
    ),
    projects: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.string,
        image: PropTypes.string,
        label: PropTypes.string
      })
    ),
    activeProjectId: PropTypes.string,
    activeAccountId: PropTypes.string,
    projectTitle: PropTypes.string,
    accountTitle: PropTypes.string,
    onProjectClick: PropTypes.func,
    onAccountClick: PropTypes.func
  };

  static defaultProps = {
    accounts: [],
    projects: []
  };

  constructor(props) {
    super(props);

    this.state = {};
  }

  toggleOpen = () => {
    this.setState({ open: !this.state.open });
  };

  handleProjectClick = id => {
    this.setState({
      activeProjectId: id,
      open: false
    });
    if (this.props.onProjectClick) {
      this.props.onProjectClick(id);
    }
  };

  handleAccountClick = id => {
    this.setState({
      activeAccountId: id,
      open: false
    });
    if (this.props.onAccountClick) {
      this.props.onAccountClick(id);
    }
  };

  activeProps() {
    const activeAccount = this.props.accounts.find(
      a => a.id === this.renderedActiveAccountId()
    );
    const activeProject = this.props.projects.find(
      a => a.id === this.renderedActiveProjectId()
    );

    if (activeAccount && activeProject) {
      const activeLabel = [activeAccount.label, activeProject.label].join(
        " / "
      );
      return {
        activeImage: activeProject.image,
        activeLabel,
        activeType: "project"
      };
    } else if (activeProject) {
      return {
        activeImage: activeProject.image,
        activeLabel: activeProject.label,
        activeType: "project"
      };
    } else if (activeAccount) {
      return {
        activeImage: activeAccount.image,
        activeLabel: activeAccount.label,
        activeType: "account"
      };
    }

    return {};
  }

  flyoutProps() {
    if (this.props.projects.length > 1 || this.props.accounts.length > 1) {
      return {
        showCaret: true,
        onClick: this.toggleOpen,
        onClickOutside: this.toggleOpen
      };
    }

    return {};
  }

  renderedActiveProjectId() {
    if (this.props.activeProjectId !== undefined) {
      return this.props.activeProjectId;
    } else if (this.state.activeProjectId !== undefined) {
      return this.state.activeProjectId;
    }
    return this.props.projects[0].id;
  }

  renderedActiveAccountId() {
    if (this.props.activeAccountId !== undefined) {
      return this.props.activeAccountId;
    } else if (this.state.activeAccountId !== undefined) {
      return this.state.activeAccountId;
    }
    return this.props.accounts[0].id;
  }

  render() {
    const projects = this.props.projects.map(project => ({
      ...project,
      active: project.id === this.renderedActiveProjectId()
    }));
    const accounts = this.props.accounts.map(account => ({
      ...account,
      active: account.id === this.renderedActiveAccountId()
    }));

    return (
      <ProjectAccountSwitcherAdapter
        open={this.state.open}
        projectTitle={this.props.projectTitle}
        accountTitle={this.props.accountTitle}
        {...this.flyoutProps()}
        {...this.activeProps()}
      >
        {accounts.map((a, i) => {
          if (i === 0) {
            return (
              <Account onClick={this.handleAccountClick} key={a.id} {...a} />
            );
          }
          return (
            <Account onClick={this.handleAccountClick} key={a.id} {...a} />
          );
        })}

        {projects.map((p, i) => {
          if (i === 0) {
            return (
              <Project onClick={this.handleProjectClick} key={p.id} {...p} />
            );
          }
          return (
            <Project onClick={this.handleProjectClick} key={p.id} {...p} />
          );
        })}
      </ProjectAccountSwitcherAdapter>
    );
  }
}

export default ProjectAccountSwitcher;
