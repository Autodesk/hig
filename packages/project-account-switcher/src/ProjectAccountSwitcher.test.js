import React from "react";
import renderer from "react-test-renderer";
import ProjectAccountSwitcher from "./ProjectAccountSwitcher";

describe("project-account-switcher/ProjectAccountSwitcher", () => {
  describe("integration", () => {
    it("renders with default account/project", () => {
      const tree = renderer
        .create(
          <ProjectAccountSwitcher
            accounts={[
              { id: "1", label: "Account 1" },
              { id: "2", label: "Account 2" },
            ]}
            accountTitle="Accounts"
            defaultAccount="1"
            defaultProject="2"
            projects={[
              { id: "1", label: "Project 1" },
              { id: "2", label: "Project 2" },
            ]}
            projectTitle="Projects"
          />
        )
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
    it("renders with controlled active account/project", () => {
      const tree = renderer
        .create(
          <ProjectAccountSwitcher
            accounts={[
              { id: "1", label: "Account 1" },
              { id: "2", label: "Account 2" },
            ]}
            accountTitle="Accounts"
            activeAccount="2"
            activeProject="1"
            defaultAccount="1"
            defaultProject="2"
            projects={[
              { id: "1", label: "Project 1" },
              { id: "2", label: "Project 2" },
            ]}
            projectTitle="Projects"
          />
        )
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
