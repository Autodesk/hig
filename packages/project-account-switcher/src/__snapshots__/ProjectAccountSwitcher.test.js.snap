// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`project-account-switcher/ProjectAccountSwitcher integration renders with controlled active account/project 1`] = `
.emotion-36 {
  position: relative;
  display: inline-block;
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-35 {
  position: absolute;
  display: none;
  width: 100%;
  z-index: 9999;
  -webkit-transition-property: opacity,-webkit-transform;
  -webkit-transition-property: opacity,transform;
  transition-property: opacity,transform;
  -webkit-transition-duration: 250ms;
  transition-duration: 250ms;
  -webkit-transition-timing-function: cubic-bezier(0.77,0,0.265,2);
  transition-timing-function: cubic-bezier(0.77,0,0.265,2);
  touch-action: auto;
  pointer-events: auto;
  opacity: 1;
  -webkit-transform: none;
  -ms-transform: none;
  transform: none;
}

.emotion-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.emotion-4 {
  margin: 0;
  padding-left: 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 34px;
  line-height: 34px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
}

.emotion-2 {
  display: inline-block;
  position: relative;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  overflow: hidden;
  height: 32px;
  width: 32px;
  border-radius: 2px;
  border: 1px solid #bec8d2;
  color: #485366;
  text-align: center;
  margin-right: 10px;
}

.emotion-0 {
  position: absolute;
  top: 0;
  left: 0;
  border: none;
  cursor: pointer;
  color: transparent;
}

.emotion-0[src=''] {
  display: none;
}

.emotion-6 {
  margin: 0 12px;
}

.emotion-1 {
  color: #3c3c3c;
  display: block;
  font-family: ArtifaktElement,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1.428571429;
  margin: 0;
  text-align: initial;
}

.emotion-5 {
  fill: #808080;
  pointer-events: none;
}

.emotion-5 > * {
  fill: #808080;
  pointer-events: none;
}

.emotion-11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: absolute;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
}

.emotion-9 {
  fill: rgba(128,128,128,0);
}

.emotion-10 {
  fill: #ffffff;
}

.emotion-32 {
  margin: 0;
  padding: 0 10px 10px 10px;
}

.emotion-34 {
  background-color: #ffffff;
  border-radius: 4px;
  border: none;
  box-shadow: 0 0 16px rgba(0,0,0,0.2);
}

.emotion-33 {
  position: relative;
}

.emotion-21 {
  margin-top: 30px;
  margin-bottom: 10px;
  list-style: none;
  padding: 0;
}

.emotion-21:first-of-type {
  margin-top: 0;
  padding-top: 10px;
}

.emotion-16 {
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 34px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  padding-top: 8px;
  white-space: nowrap;
}

.emotion-14 {
  display: inline-block;
  position: relative;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  overflow: hidden;
  height: 30px;
  width: 60px;
  border-radius: 2px;
  border: 1px solid #bec8d2;
  color: #485366;
  text-align: center;
  margin-right: 10px;
}

.emotion-15 {
  font-weight: 400;
  font-size: 14px;
  line-height: 1.428571429;
  font-family: ArtifaktElement,sans-serif;
  margin: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  color: #006eaf;
  outline: none;
}

<div
  className="emotion-36"
>
  <div
    className="emotion-8"
  >
    <div
      className="emotion-7"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      <div
        className="emotion-4"
      >
        <span
          className="emotion-2"
        >
          <img
            alt="Account 2 / Project 1"
            className="emotion-0"
            src=""
          />
          <span
            className="emotion-1"
            style={
              Object {
                "lineHeight": "32px",
                "textAlign": "center",
              }
            }
          >
            A2/P1
          </span>
        </span>
        <p
          className="emotion-1"
        >
          Account 2 / Project 1
        </p>
      </div>
      <div
        className="emotion-6"
      >
        <svg
          className="emotion-5"
          height="10px"
          viewBox="0 0 10 10"
          width="10px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            caret-down-m
          </title>
          <path
            d="M8.71 3.22a.75.75 0 0 0-1.06 0L5 5.87 2.35 3.22a.75.75 0 0 0-1.06 0 .74.74 0 0 0 0 1.06l3.36 3.36a.5.5 0 0 0 .7 0l3.36-3.36a.74.74 0 0 0 0-1.06z"
          />
        </svg>
      </div>
    </div>
  </div>
  <div
    className="emotion-35"
    style={
      Object {
        "left": "0px",
        "top": "0px",
      }
    }
  >
    <div
      aria-hidden="true"
      className="emotion-11"
      role="presentation"
      style={
        Object {
          "left": "0px",
          "top": "0px",
        }
      }
    >
      <svg
        height="24"
        viewBox="0 0 24 24"
        width="24"
      >
        <polygon
          className="emotion-9"
          points="0,12 12,0 24,12"
        />
        <polygon
          className="emotion-10"
          points="2.5,12 12,2.5 21.5,12"
        />
      </svg>
    </div>
    <div
      className="emotion-34"
      style={
        Object {
          "maxHeight": undefined,
        }
      }
    >
      <div
        className="emotion-33"
      >
        <div
          className="emotion-32"
        >
          <div>
            <ul
              className="emotion-21"
            >
              <span
                className="emotion-1"
                style={
                  Object {
                    "fontSize": "11px",
                    "textTransform": "uppercase",
                  }
                }
              >
                Accounts
              </span>
              <li
                className="emotion-16"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="menuitem"
                tabIndex="0"
              >
                <span
                  className="emotion-14"
                >
                  <span
                    className="emotion-1"
                    style={
                      Object {
                        "lineHeight": "32px",
                        "textAlign": "center",
                      }
                    }
                  >
                    A1
                  </span>
                </span>
                <span
                  className="emotion-15"
                  onBlur={[Function]}
                  onClick={undefined}
                  onFocus={[Function]}
                  onMouseDown={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  role="button"
                >
                  Account 1
                </span>
              </li>
              <li
                className="emotion-16"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="menuitem"
                tabIndex="0"
              >
                <span
                  className="emotion-14"
                >
                  <span
                    className="emotion-1"
                    style={
                      Object {
                        "lineHeight": "32px",
                        "textAlign": "center",
                      }
                    }
                  >
                    A2
                  </span>
                </span>
                <span
                  className="emotion-15"
                  onBlur={[Function]}
                  onClick={undefined}
                  onFocus={[Function]}
                  onMouseDown={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  role="button"
                >
                  Account 2
                </span>
              </li>
            </ul>
            <ul
              className="emotion-21"
            >
              <span
                className="emotion-1"
                style={
                  Object {
                    "fontSize": "11px",
                    "textTransform": "uppercase",
                  }
                }
              >
                Projects
              </span>
              <li
                className="emotion-16"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="menuitem"
                tabIndex="0"
              >
                <span
                  className="emotion-2"
                >
                  <span
                    className="emotion-1"
                    style={
                      Object {
                        "lineHeight": "32px",
                        "textAlign": "center",
                      }
                    }
                  >
                    P1
                  </span>
                </span>
                <span
                  className="emotion-15"
                  onBlur={[Function]}
                  onClick={undefined}
                  onFocus={[Function]}
                  onMouseDown={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  role="button"
                >
                  Project 1
                </span>
              </li>
              <li
                className="emotion-16"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="menuitem"
                tabIndex="0"
              >
                <span
                  className="emotion-2"
                >
                  <span
                    className="emotion-1"
                    style={
                      Object {
                        "lineHeight": "32px",
                        "textAlign": "center",
                      }
                    }
                  >
                    P2
                  </span>
                </span>
                <span
                  className="emotion-15"
                  onBlur={[Function]}
                  onClick={undefined}
                  onFocus={[Function]}
                  onMouseDown={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  role="button"
                >
                  Project 2
                </span>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`project-account-switcher/ProjectAccountSwitcher integration renders with default account/project 1`] = `
.emotion-36 {
  position: relative;
  display: inline-block;
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-35 {
  position: absolute;
  display: none;
  width: 100%;
  z-index: 9999;
  -webkit-transition-property: opacity,-webkit-transform;
  -webkit-transition-property: opacity,transform;
  transition-property: opacity,transform;
  -webkit-transition-duration: 250ms;
  transition-duration: 250ms;
  -webkit-transition-timing-function: cubic-bezier(0.77,0,0.265,2);
  transition-timing-function: cubic-bezier(0.77,0,0.265,2);
  touch-action: auto;
  pointer-events: auto;
  opacity: 1;
  -webkit-transform: none;
  -ms-transform: none;
  transform: none;
}

.emotion-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.emotion-4 {
  margin: 0;
  padding-left: 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 34px;
  line-height: 34px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
}

.emotion-2 {
  display: inline-block;
  position: relative;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  overflow: hidden;
  height: 32px;
  width: 32px;
  border-radius: 2px;
  border: 1px solid #bec8d2;
  color: #485366;
  text-align: center;
  margin-right: 10px;
}

.emotion-0 {
  position: absolute;
  top: 0;
  left: 0;
  border: none;
  cursor: pointer;
  color: transparent;
}

.emotion-0[src=''] {
  display: none;
}

.emotion-6 {
  margin: 0 12px;
}

.emotion-1 {
  color: #3c3c3c;
  display: block;
  font-family: ArtifaktElement,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1.428571429;
  margin: 0;
  text-align: initial;
}

.emotion-5 {
  fill: #808080;
  pointer-events: none;
}

.emotion-5 > * {
  fill: #808080;
  pointer-events: none;
}

.emotion-11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: absolute;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
}

.emotion-9 {
  fill: rgba(128,128,128,0);
}

.emotion-10 {
  fill: #ffffff;
}

.emotion-32 {
  margin: 0;
  padding: 0 10px 10px 10px;
}

.emotion-34 {
  background-color: #ffffff;
  border-radius: 4px;
  border: none;
  box-shadow: 0 0 16px rgba(0,0,0,0.2);
}

.emotion-33 {
  position: relative;
}

.emotion-21 {
  margin-top: 30px;
  margin-bottom: 10px;
  list-style: none;
  padding: 0;
}

.emotion-21:first-of-type {
  margin-top: 0;
  padding-top: 10px;
}

.emotion-16 {
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 34px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  padding-top: 8px;
  white-space: nowrap;
}

.emotion-14 {
  display: inline-block;
  position: relative;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  overflow: hidden;
  height: 30px;
  width: 60px;
  border-radius: 2px;
  border: 1px solid #bec8d2;
  color: #485366;
  text-align: center;
  margin-right: 10px;
}

.emotion-15 {
  font-weight: 400;
  font-size: 14px;
  line-height: 1.428571429;
  font-family: ArtifaktElement,sans-serif;
  margin: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  color: #006eaf;
  outline: none;
}

<div
  className="emotion-36"
>
  <div
    className="emotion-8"
  >
    <div
      className="emotion-7"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      <div
        className="emotion-4"
      >
        <span
          className="emotion-2"
        >
          <img
            alt="Account 1 / Project 2"
            className="emotion-0"
            src=""
          />
          <span
            className="emotion-1"
            style={
              Object {
                "lineHeight": "32px",
                "textAlign": "center",
              }
            }
          >
            A1/P2
          </span>
        </span>
        <p
          className="emotion-1"
        >
          Account 1 / Project 2
        </p>
      </div>
      <div
        className="emotion-6"
      >
        <svg
          className="emotion-5"
          height="10px"
          viewBox="0 0 10 10"
          width="10px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            caret-down-m
          </title>
          <path
            d="M8.71 3.22a.75.75 0 0 0-1.06 0L5 5.87 2.35 3.22a.75.75 0 0 0-1.06 0 .74.74 0 0 0 0 1.06l3.36 3.36a.5.5 0 0 0 .7 0l3.36-3.36a.74.74 0 0 0 0-1.06z"
          />
        </svg>
      </div>
    </div>
  </div>
  <div
    className="emotion-35"
    style={
      Object {
        "left": "0px",
        "top": "0px",
      }
    }
  >
    <div
      aria-hidden="true"
      className="emotion-11"
      role="presentation"
      style={
        Object {
          "left": "0px",
          "top": "0px",
        }
      }
    >
      <svg
        height="24"
        viewBox="0 0 24 24"
        width="24"
      >
        <polygon
          className="emotion-9"
          points="0,12 12,0 24,12"
        />
        <polygon
          className="emotion-10"
          points="2.5,12 12,2.5 21.5,12"
        />
      </svg>
    </div>
    <div
      className="emotion-34"
      style={
        Object {
          "maxHeight": undefined,
        }
      }
    >
      <div
        className="emotion-33"
      >
        <div
          className="emotion-32"
        >
          <div>
            <ul
              className="emotion-21"
            >
              <span
                className="emotion-1"
                style={
                  Object {
                    "fontSize": "11px",
                    "textTransform": "uppercase",
                  }
                }
              >
                Accounts
              </span>
              <li
                className="emotion-16"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="menuitem"
                tabIndex="0"
              >
                <span
                  className="emotion-14"
                >
                  <span
                    className="emotion-1"
                    style={
                      Object {
                        "lineHeight": "32px",
                        "textAlign": "center",
                      }
                    }
                  >
                    A1
                  </span>
                </span>
                <span
                  className="emotion-15"
                  onBlur={[Function]}
                  onClick={undefined}
                  onFocus={[Function]}
                  onMouseDown={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  role="button"
                >
                  Account 1
                </span>
              </li>
              <li
                className="emotion-16"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="menuitem"
                tabIndex="0"
              >
                <span
                  className="emotion-14"
                >
                  <span
                    className="emotion-1"
                    style={
                      Object {
                        "lineHeight": "32px",
                        "textAlign": "center",
                      }
                    }
                  >
                    A2
                  </span>
                </span>
                <span
                  className="emotion-15"
                  onBlur={[Function]}
                  onClick={undefined}
                  onFocus={[Function]}
                  onMouseDown={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  role="button"
                >
                  Account 2
                </span>
              </li>
            </ul>
            <ul
              className="emotion-21"
            >
              <span
                className="emotion-1"
                style={
                  Object {
                    "fontSize": "11px",
                    "textTransform": "uppercase",
                  }
                }
              >
                Projects
              </span>
              <li
                className="emotion-16"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="menuitem"
                tabIndex="0"
              >
                <span
                  className="emotion-2"
                >
                  <span
                    className="emotion-1"
                    style={
                      Object {
                        "lineHeight": "32px",
                        "textAlign": "center",
                      }
                    }
                  >
                    P1
                  </span>
                </span>
                <span
                  className="emotion-15"
                  onBlur={[Function]}
                  onClick={undefined}
                  onFocus={[Function]}
                  onMouseDown={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  role="button"
                >
                  Project 1
                </span>
              </li>
              <li
                className="emotion-16"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="menuitem"
                tabIndex="0"
              >
                <span
                  className="emotion-2"
                >
                  <span
                    className="emotion-1"
                    style={
                      Object {
                        "lineHeight": "32px",
                        "textAlign": "center",
                      }
                    }
                  >
                    P2
                  </span>
                </span>
                <span
                  className="emotion-15"
                  onBlur={[Function]}
                  onClick={undefined}
                  onFocus={[Function]}
                  onMouseDown={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  role="button"
                >
                  Project 2
                </span>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
